service/src/main/java/org/mskcc/smile/service/impl/CorrectCmoPatientHandlingServiceImpl.java

```diff
@@ -192,26 +193,25 @@ private void setupCorrectCmoPatientIdHandler(Gateway gateway,
             public void onMessage(Message msg, Object message) {
                 LOG.info("Received message on topic: " + CORRECT_CMOPTID_TOPIC);
                 Map<String, String> incomingDataMap = new HashMap<>();
+                String oldCmoPatientId = null;
+                String newCmoPatientId = null;
                 try {
-                    // do not log contents of incoming message
                     String incomingDataString = mapper.readValue(
                             new String(msg.getData(), StandardCharsets.UTF_8),
                             String.class);
                     incomingDataMap = mapper.readValue(incomingDataString, Map.class);
-                } catch (JsonProcessingException e) {
-                    LOG.error("Error processing the incoming data map. "
-                            + "Refer to NATS logs for more details.");
+                    oldCmoPatientId = crdbMappingService.getCmoPatientIdByInputId(
```

Comment:
From code, I understand these additions and what is happening in onMessage(), but I don't understand the reasoning.  For example, after attempting to set oldCmoPatientId here, it is then set on line [224](https://github.com/mskcc/smile-server/blob/95b89a6943b4697459785285a2dc5adfe6327d35/service/src/main/java/org/mskcc/smile/service/impl/CorrectCmoPatientHandlingServiceImpl.java#L224C29-L224C76) if the oldCmoPatientId was not found in CRDB, but does start with a "C-".  This is the same for newCmoPatientId.  If a CMO patient id is not found in CRDB, should we be processing it?

Reply:
Yeah we decided that we would still be processing these ID's. CRDB lookup table only returns the barcode without the leading "C-"